src/v3/src/transformer/transaction/transform.ts

```diff
@@ -44,6 +50,41 @@ const transformRemediationNameForIdp: TransformStepFnWithOptions = (options) =>
   return formbag;
 };
 
+const transformRegistrationSchema: TransformStepFnWithOptions = (options) => (formbag) => {
+  const {
+    transaction: { nextStep },
+    widgetProps,
+    setMessage,
+    isClientTransaction,
+  } = options;
+  if (nextStep?.name !== IDX_STEP.ENROLL_PROFILE) {
+    return formbag;
+  }
+
+  const inputs = nextStep.inputs!;
+  // Converts IDX Input array into Registration Schema element array
+  const registrationSchema = convertIdxInputsToRegistrationSchema(inputs);
+
+  parseRegistrationSchema(
+    widgetProps,
+    registrationSchema,
+    (schema: RegistrationElementSchema[]) => {
+      // Converts Registration Schema Element array back into IDX Input array
+      // Also sets those inputs back into the nextStep.inputs property
+      convertRegistrationSchemaToIdxInputs(schema, inputs);
+    },
+    (error: APIError) => {
+      // Prioritize server side errors over custom error message
+      if (isClientTransaction) {
+        return;
+      }
+      triggerRegistrationErrorMessages(error, inputs, setMessage);
+    },
+  );
+  return formbag;
+};
```

Comment:
This is the meat of the parseSchema functionality.

TLDR;

I am taking the inputs sent to us from the API response and converting them into the Registration schema format (which is what v2 is currently expecting). Then, I am passing it to the callback (to allow the customer to add/update/etc) the schema, and in the body of the callback, I execute a function that converts the modified schema back into the IDX `Input` object array to allow it to pass through the SIW standard transformation functions (where we will get things like validation and element creation for free).